pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -230,7 +230,7 @@ public static String arrayElementAtString(String[] arr, int idx) {
 
   @ScalarFunction(names = {"array", "arrayValueConstructor"}, isVarArg = true)
   public static Object arrayValueConstructor(Object... arr) {
-    if (arr.length == 0) {
+    if (arr == null || arr.length == 0 || arr[0] == null) {
```

Comment:
Firstly, to give the context:
foo(String... args) {
args == null // foo(null)
args.length == 0 // foo()
args[0] == null // String bar = null; foo(bar)

In this specific case,  if arr[0] == null, the code below (`arr[0].getClass()`) will cause null pointer exception, hence the additional checks is an improvement. I didn't follow why it would change the semantics.

I have tried to only add the additional checks in places which can cause NPE, vs changing all varargs usages in the codebase.

Reply:
It changes semantics because `foo(null, 1, 2, 3)` will behave differently `foo(1, 2, 3, null)`. Do we want to enter on this if when _any_ value in `arr` is null? Ok, but doing so only when the _first_ value is null looks very strange.

Reply:
`foo(null, 1, 2, 3)` will cause NPE below (`arr[0].getClass()`), are we ok with that? This PR address that.
`if (arr == null || arr.length == 0)` check will not prevent the NPE.

Reply:
My point is that `foo(1, null, 2, 3)` will also cause a NPE when doing `intArr[i] = (Integer) arr[i];`.

It is very strange to do not throw NPE when the first element is null but do it when the second is. We should either always throw NPE when any value in the array is null (like before) or never do that

Reply:
Correct. I missed `(Integer) arr[i]` code, let me modify the PR.